tubesml/encoders.py

```diff
@@ -56,24 +58,28 @@ def fit(self, X, y):
         :param y: array-like of shape (n_samples,) or (n_samples, n_outputs).
             The target values (or class labels) as integers or floats.
         """
-        if self.agg_func == "count":
-            raise UserWarning("Frequency encoding not supported")  # TODO: allow this in the future
+        
         # Encode all categorical cols by default
         if self.to_encode is None:
             self.to_encode = [c for c in X if str(X[c].dtype) == "object" or str(X[c].dtype) == "category"]
 
         tmp = X.copy()
         tmp["target"] = y
 
-        self.prior_ = tmp["target"].agg(self.agg_func)
-        self.posteriors_ = {}
-
-        for col in self.to_encode:
-            agg = tmp.groupby(col)["target"].agg(["count", self.agg_func])
-            counts = agg["count"]
-            data = agg[self.agg_func]
-            pw = self.prior_weight
-            self.posteriors_[col] = ((pw * self.prior_ + counts * data) / (pw + counts)).to_dict()
+        if self.agg_func == "count":
+            self.counts_ = {}
+            for col in self.to_encode:
+                counts = tmp[col].value_counts()
+                self.counts_[col] = collections.defaultdict(lambda: 0, counts.to_dict())
```

Comment:
Elegant, I like it

Reply:
The only objection I have is that, to keep the logic consistent with the other options, you would create `self.prior_` to be the default dictionary with all 0s, then `self.posteriors_` as an update of the prior dictionary. This would avoid the if-statement in the transform method. 

That being said, it is not a hill I want ot die on and I am fully open to the idea I am nitpicking :)